tensilelite/Tensile/Components/SIA.py

```diff
@@ -170,13 +170,39 @@ def getLocalWriteMFMAEnd(writer, kernel, tensorParametersA, tensorParametersB):
     isLocalReadsOpt = False
     tmpLatencyLeft  = 0
     tmpNumMfmaForLR = 0
-    localReadsNotWaited = writer.states.numReadsPerIterB//kernel["InnerUnroll"] - writer.states.numReadsPerUnrollB
+    localReadsNotWaited = writer.states.numReadsPerIterB // kernel["InnerUnroll"] - writer.states.numReadsPerUnrollB
     if kernel["UnrollMajorLDSB"] and localReadsNotWaited > 0:
         isLocalReadsOpt = True
 
+    # Add space to avoid LR FIFO stall
+    # lrStallLatencyBuffer:
+    # 40 quad-cycle - 4 x miLatency for b128
+    # 20 quad-cycle - 4 x miLatency for b64 (equal to one miLatency)
+    # 10 quad-cycle - 4 x miLatency for b32 (no stall)
+    # so no stall happen for b64/b32/b16
+    localReadFIFO = []
+    def checkLocalReadFIFO(currentMFMA, blockWidth):
```

Comment:
could we declare it as global function to prevent duplicate the code?

Reply:
yes. done. please review again